test/test_dirent.rs

```diff
@@ -0,0 +1,42 @@
+use nix::dirent::{self, opendir, readdir, seekdir, telldir, Dir};
+use std::path::Path;
+use tempdir::TempDir;
+
+fn test_readdir<OPEN>(open_fn: OPEN)
+    where OPEN: Fn(&Path) -> Dir
+{
+    let tempdir = TempDir::new("nix-test_readdir")
+        .unwrap_or_else(|e| panic!("tempdir failed: {}", e));
+    let mut dir = open_fn(tempdir.path());
+    let letter1 = readdir(&mut dir)
+                       .unwrap()
+                       .unwrap()
+                       .name()
+                       .to_bytes()[0];
+    assert_eq!(letter1, '.' as u8);
+
+    let pos = telldir(&mut dir);
+    seekdir(&mut dir, pos); // no-op
```

Comment:
Could we have a test where we seek back a directory and check that the same directory is read twice (so that we no `seekdir` worked? I'd like to have a non-nop test of it.